hugegraph-core/src/main/java/org/apache/hugegraph/backend/tx/EphemeralJobQueue.java

```diff
@@ -0,0 +1,237 @@
+                }
+
+                queue = this.queueWeakReference.get();
+                if (queue == null) {
+                    stop = true;
+                    continue;
+                }
+
+                if (queue.isEmpty() || consumeCount > MAX_CONSUME_COUNT ||
+                    interruptedException != null) {
+                    queue.consumeComplete();
+                    stop = true;
+                    if (!queue.isEmpty()) {
+                        queue.reScheduleIfNeeded();
+                    }
+                    continue;
+                }
+
+                try {
+                    while (!queue.isEmpty() && batchJobs.size() < PAGE_SIZE) {
+                        EphemeralJob<?> job = queue.queue().poll();
+                        if (job == null) {
+                            continue;
+                        }
+                        batchJobs.add(job);
+                        consumeCount++;
+                    }
+
+                    if (batchJobs.isEmpty()) {
+                        continue;
+                    }
+
+                    ret = this.executeBatchJob(batchJobs, ret);
+
+                } catch (InterruptedException e) {
+                    interruptedException = e;
+                } finally {
+                    batchJobs.clear();
+                }
+            }
+
+            if (interruptedException != null) {
+                Thread.currentThread().interrupt();
+                throw interruptedException;
+            }
+
+            return ret;
+        }
+
+        private Object executeBatchJob(List<EphemeralJob<?>> jobs, Object prevResult) throws Exception {
+            GraphIndexTransaction graphTx = this.params().systemTransaction().indexTransaction();
+            GraphIndexTransaction systemTx = this.params().graphTransaction().indexTransaction();
+            Object result = prevResult;
+            for (EphemeralJob<?> job : jobs) {
+                this.initJob(job);
+                Object obj = job.call();
+                if (job instanceof Reduce) {
+                    result = ((Reduce) job).reduce(result, obj);
+                }
+            }
+
+            graphTx.commit();
+            systemTx.commit();
+
+            return result;
+        }
+
+        private void initJob(EphemeralJob<?> job) {
+            job.graph(this.graph());
+            job.params(this.params());
+        }
+
+        @Override
+        public Object call() throws Exception {
+            try {
+                return super.call();
+            } catch (Throwable e) {
+                LOG.warn("Failed to execute BatchEphemeralJob", e);
+                EphemeralJobQueue queue = this.queueWeakReference.get();
+                if (e instanceof InterruptedException) {
+                    Thread.currentThread().interrupt();
+                    if (queue != null) {
+                        queue.queue().clear();
```

Comment:
not sure why called clear() here.
and prefer to call `queue.clear()`.

Reply:
when the interruptedException occur, perhaps these tasks aborted, so clear all.